packages/material/src/mui-controls/MuiSelect.tsx

```diff
@@ -49,11 +50,11 @@ export const MuiSelect = React.memo((props: EnumCellProps & WithClassname) => {
       disabled={!enabled}
       autoFocus={appliedUiSchemaOptions.focus}
       value={data !== undefined ? data : ''}
-      onChange={ev => handleChange(path, ev.target.value)}
+      onChange={ev =>handleChange(path, ev.target.value === '' ? undefined : ev.target.value)}
```

Comment:
This will delete values which are empty on purpose. We really need to check whether this was our `None` entry.

Reply:
On further inspection it seems it's not straightforward to get this information out of Material UI's `Select`.